src/script/script.js

```diff
@@ -0,0 +1,402 @@
+          <span class="caseUp hidden">${obj.ru.up}</span>
+          <span class="caps hidden">${obj.ru.caps}</span>
+          <span class="shiftCaps hidden">${obj.ru.shiftcaps}</span>
+        </span>
+        <span class="en ${localStorage.keyboardLanguage === "en" ? "" : "hidden"}">
+          <span class="caseDown">${obj.en.down}</span>
+          <span class="caseUp hidden">${obj.en.up}</span>
+          <span class="caps hidden">${obj.en.caps}</span>
+          <span class="shiftCaps hidden">${obj.en.shiftcaps}</span>
+        </span>`;
+    btn.innerHTML = output;
+    return btn;
+  }
+
+  create() {
+    this.generateHeader();
+    this.generateTextArea();
+    this.generateKeyboard();
+  }
+
+  remove() {
+    this.rootElement.remove();
+  }
+
+  ///////////////////////////////////////////////////////////////
+  switchLanguage() {
+    localStorage.keyboardLanguage = localStorage.keyboardLanguage === "ru" ? "en" : "ru";
+    for (let key of this.keys) {
+      let items = key.children;
+      Array.from(items).forEach(el => {
+        if (el.classList.contains(localStorage.keyboardLanguage)) {
+          el.classList.remove('hidden');
+          this.toggleClassesInButton(key, "caseDown");
+        } else {
+          el.classList.add('hidden');
+        }
+      });
+    }
+  }
+
+  textAreaChangeInfo(value) {
+    let text = this.textArea.value;
+    this.textArea.focus();
+    if (value === "&amp;"){
+      this.textArea.setRangeText("&", this.textArea.selectionStart, this.textArea.selectionEnd, "end");
+    } else if (value === "Backspace") {
+      if (this.currentPosition) {
+        this.textArea.value = text.slice(0, this.currentPosition - 1) + '' + text.slice(this.currentPosition, text.length);
+        this.textArea.setRangeText("", this.currentPosition, this.currentPosition, "end");
+      } else {
+        this.textArea.value = text.slice(0, -1);
+      }
+    } else if (value === "Delete"){
+      this.textArea.value = text.slice(0, this.currentPosition) + '' + text.slice(this.currentPosition+1, text.length);
+      this.textArea.setRangeText("", this.currentPosition, this.currentPosition, "end");
+    } else {
+      this.textArea.setRangeText(value, this.textArea.selectionStart, this.textArea.selectionEnd, "end");
+    }
+  }
+
+  handleButtonDown(eventCode) {
+    if (this.exceptions.includes(eventCode)) return;
+    let clickedButton = document.querySelector(`.${eventCode}`);
+    if (eventCode === "CapsLock") {
+      clickedButton.classList.toggle('active');
+      this.capslocked = !this.capslocked;
+      this.toggleCappslock();
+      return;
+    }
+    if (!this.pressed.has(eventCode)) {
+      this.pressed.add(eventCode);
+      clickedButton.classList.toggle('active');
+    }
+
+    if (clickedButton.classList.contains('symbol')) {
+      this.textAreaChangeInfo(clickedButton.dataset.value);
+    } else {
+      if (eventCode === "ShiftRight" || eventCode === "ShiftLeft") {
```

Comment:
1. In cases like that you could use switch case
2. You doesn`t need empty return state. It makes no sense

Reply:
In this case, I cannot use the switch case construct, because I check different data - in some cases, is there a value in this.pressedButtons, and in some cases I check the value itself.

Reply:
completely refactored